(feat): Add Positional Trait in Go codegen

rishav-karanjit opened the pull request.

rishav-karanjit committed:
Add runtime (remove later)

rishav-karanjit committed:
Comment all test model for faster workflow run

rishav-karanjit committed:
Add runtime

rishav-karanjit committed:
Changes for the positional trait

rishav-karanjit committed:
Changes for positional trait

rishav-karanjit committed:
Remove redundant code

rishav-karanjit committed:
Add everything

rishav-karanjit committed:
Revert the unwanted changes

rishav-karanjit committed:
Changes for shim

rishav-karanjit committed:
Update API client return type for interface

rishav-karanjit committed:
maybe output to output

rishav-karanjit committed:
get dafny type from DafnyNameResolver

rishav-karanjit committed:
remove print statement

rishav-karanjit committed:
api client part where we return values

rishav-karanjit committed:
Uncomment so that CI runs

rishav-karanjit committed:
Remove runtimes

rishav-karanjit committed:
Add go.mod

rishav-karanjit committed:
remove dafny codegen files

rishav-karanjit committed:
Update go mod

rishav-karanjit committed:
Fix symbolForPositional

rishav-karanjit committed:
Update to DafnyNameResolver

rishav-karanjit committed:
Update getSmithyType

rishav-karanjit committed:
Replace with isOptional

rishav-karanjit committed:
Replaced isOptional with isPointable

rishav-karanjit committed:
Update DafnyNameResolver

rishav-karanjit committed:
Update DafnyNameResolver

rishav-karanjit committed:
Copy GoPointableIndex from rishav-union

rishav-karanjit committed:
Merge with scchatur/GoCodegen

ShubhamChaturvedi7 commented on codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/knowledge/GoPointableIndex.java:
```diff
@@ -15,263 +15,270 @@
-        return dereferencableShapes.contains(shape.toShapeId());
-    }
-}
+ package software.amazon.polymorph.smithygo.codegen.knowledge;
```
What happened here? Why the whole file is in the diff??

rishav-karanjit replied:
This diff will go away once I merge new changes to this PR.

ShubhamChaturvedi7 commented on codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java:
```diff
@@ -141,10 +144,20 @@ func NewClient(clientConfig $L) (*$T, error) {
             } else {
+                String dafnyType; 
+                if (inputShape.hasTrait(PositionalTrait.class)) {
+                    Shape inputForPositional = model.expectShape(inputShape.getAllMembers().values().stream().findFirst().get().getTarget());
```
Why not `getAllMembers().get(0)` assuming it's a collection?

rishav-karanjit replied:
`getAllMembers()` is a map but `getAllMembers().values()` is a collection. However, collection does not have `get` method.

ShubhamChaturvedi7 commented on codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java:
```diff
@@ -238,13 +260,26 @@ void generateShim() {
                     clientResponse = "var native_error";
                     returnResponse = "dafny.TupleOf()";
                     writer.addImportFromModule("github.com/dafny-lang/DafnyRuntimeGo", "dafny");
+                    inputType = maybeInputType;
```
Q: Why inputType is being assigned on a conditional on outputType?

ShubhamChaturvedi7 commented on codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java:
```diff
@@ -32,6 +34,10 @@ public static String shapeNamespace(final Shape shape) {
     }
 
     public static String smithyTypesNamespace(final Shape shape) {
+        // if (shape.isResourceShape()) {   
+        //     // TypesNamespace for resources are in ShapeName        
+        //     return shape.getId().getName();
+        // }
```
Remove the commented code.

rishav-karanjit committed:
Put check for PositionalTrait outside outputshape check

rishav-karanjit committed:
Remove comments

rishav-karanjit committed:
Merge from scchatur/GoCodegen

rishav-karanjit committed:
Merge fix

rishav-karanjit committed:
Merge branch 'scchatur/GoCodegen' into rishav-positional

alex-chew committed:
feat: Rust codegen for more SimpleTypes models (#542)

ajewellamz committed:
fix: bump Rust version to 1.80 (#553)

* fix: bump Rust version to 1.80

rishav-karanjit committed:
Remove unwanted whitespace

robin-aws committed:
chore(ci): make all pull.yml jobs required in branch protection automatically (#554)

As is the case for many suites of GitHub actions, it's a huge pain to manually maintain the list of required checks in the branch protection. This change leverages https://github.com/re-actors/alls-green to add a single capstone action that only passes if all dependent jobs pass (see repo README for details on why the naive approach doesn't work).

Once this is approved I'll update the required checks before merging.

Also adding Python tests to the list of triggers in the nightly build since I noticed it was missing.

josecorella committed:
chore: add Dafny 4.8.0 (#522)

ShubhamChaturvedi7 committed:
fix: CI

ShubhamChaturvedi7 committed:
Merge branch 'main-1.x' into Golang/dev

ShubhamChaturvedi7 committed:
fix: use gradle for CI

ShubhamChaturvedi7 committed:
fix: Add CI Tests

ShubhamChaturvedi7 committed:
fix: Drop the removeDots script

ShubhamChaturvedi7 committed:
fix: prettify

rishav-karanjit committed:
Merge branch 'Golang/dev' into rishav-positional

rishav-karanjit committed:
Revert "Merge branch 'Golang/dev' into rishav-positional"

This reverts commit 61d7284d9495a08ba5284a540ca1e4ed8022479b, reversing
changes made to 0a707a2e66fc492740d6a2c9b5b83eb0b569eca8.

rishav-karanjit committed:
Revert "Revert "Merge branch 'Golang/dev' into rishav-positional""

This reverts commit ca4d375e21c76aa4dc2a45bd734670b5e8e0961c.

rishav-karanjit committed:
Merge fix structure generator

rishav-karanjit committed:
Format
